server/task_boltdb.go

```diff
@@ -221,32 +225,39 @@ func loadTaskLogs(tx *bolt.Tx, task *tes.Task) {
 // GetTask gets a task, which describes a running task
 func (taskBolt *TaskBolt) GetTask(ctx context.Context, req *tes.GetTaskRequest) (*tes.Task, error) {
 	var task *tes.Task
-	err := taskBolt.db.View(func(tx *bolt.Tx) error {
-		task = getTaskView(tx, req.Id, req.View)
-		return nil
+	var err error
+	err = taskBolt.db.View(func(tx *bolt.Tx) error {
+		task, err = getTaskView(tx, req.Id, req.View)
+		return err
 	})
-	return task, err
+	if err != nil {
```

Comment:
Right now that is the only type of error that can occur, but that may not always be true. Good points.